services/consensusalgo/benchmarkconsensus/test/context.go

```diff
@@ -0,0 +1,68 @@
+package test
+
+import (
+	"github.com/orbs-network/go-mock"
+	"github.com/orbs-network/orbs-network-go/config"
+	"github.com/orbs-network/orbs-network-go/instrumentation"
+	"github.com/orbs-network/orbs-network-go/services/consensusalgo/benchmarkconsensus"
+	testInstrumentation "github.com/orbs-network/orbs-network-go/test/harness/instrumentation"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/consensus"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+)
+
+type context struct {
+	gossip           *gossiptopics.MockBenchmarkConsensus
+	blockStorage     *services.MockBlockStorage
+	consensusContext *services.MockConsensusContext
+	reporting        instrumentation.Reporting
+	config           benchmarkconsensus.Config
+	service          services.ConsensusAlgoBenchmark
+}
+
+func newContext(
+	isLeader bool,
+) *context {
+
+	leaderPublicKey := []byte{0x01}
+	nodePublicKey := leaderPublicKey
+	if !isLeader {
+		nodePublicKey = []byte{0x02}
+	}
+
+	config := config.NewHardCodedConfig(
```

Comment:
Please don't instantiate `HardCodecConfig` here; the consensus algo needs a subset of that config so better to implement a stub struct in the test to implement that interface.

Reply:
Yeah.. but this is super annoying. Because I need to implement pretty much HardcodedConfig from scratch. We need some way to create partials of HardcodedConfig.. Maybe create it using a struct instead of a ctor and then I can pass only what I want to initialize and it will initialize the rest to nil. I don't want every component to have a boilerplate implementation similar to HardcodedConfig.